modules/core/src/main/java/org/apache/ignite/internal/processors/query/schema/SchemaIndexCacheVisitorImpl.java

```diff
@@ -157,8 +171,55 @@ public SchemaIndexCacheVisitorImpl(GridCacheContext cctx, SchemaIndexCacheFilter
             throw e;
         }
 
-        if (fut != null)
-            fut.get();
+        if (fut != null && stat != null) {
```

Comment:
Before the change we were waiting complete compound future, now we can skip it. It looks strange.